src/App.js

```diff
@@ -0,0 +1,227 @@
+import React, {Component} from 'react';
+import FormContainer from './containers/FormContainer/FormContainer';
+import {BrowserRouter} from 'react-router-dom';
+
+class App extends Component {
+    constructor() {
+        super();
+        this.state = {
+            todos: [
+                // {id:1 content:"Task1", done:false}
+            ]
+        }
+
+
+    }
+
+    // fetching todos from db.json
+    fetchData = (url) => {
+        fetch(url)
+            .then((response) => {
+                console.log(response);
+                return response
+            })
+            .then((response) => response.json())
+            .then((todos) => this.setState({todos: todos}))
+            .catch((error) => console.error(error));
+    };
+
+    //GET TODOS
+
+    fetchDataHandler = () => {
+        this.fetchData('http://localhost:3000/todos')
+    };
+
+
+    componentWillMount = () => {
+        this.fetchData('http://localhost:3000/todos');
+        console.log('component will mount');
+        console.log(this.state.todos);
+    };
+
+
+    // these method belows are used for CRUD of todos on json-server, not state.
+
+    //POST
+
+    postTodoHandler = (content_input) => {
+        fetch('http://localhost:3000/todos', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                'content': content_input,
+                'done': false
+            })
+        }).then(res => res.json())
+            .then(console.log())
+            .then(() => this.fetchDataHandler());
+    };
+
+    //PUT
+
+    putTodoContentHandler = (event, todo_id, done_stats) => {
+        const todoIndex = this.state.todos.findIndex(todo => {
+            return todo.id === todo_id;
+        });
+        const updatedTodo = {
+            ...this.state.todos[todoIndex]
+        };
+
+        updatedTodo.content = event.target.value;
+
+        const todos = [...this.state.todos];
+        todos[todoIndex] = updatedTodo;
+
+        this.setState({todos: todos});
+
+        let updated_content = event.target.value;
+
+        fetch('http://localhost:3000/todos/' + todo_id, {
+            method: 'PUT',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                'content': updated_content,
+                'done': done_stats
+            })
+        }).then(res => res.json())
+            .then(console.log)
+            .then(() => this.fetchDataHandler());
+    };
+
+
+    putTodoStatusHandler = (todo_id, current_content, done_stats) => {
```

Comment:
`putTodoStatusHandler`  and `putTodoContentHandler` can be a same function, if you pass the same todo object.

Reply:
I use putTodoStatusHandler in checkbox, for onChange.  And at the same time, I use putTodoContentHandler in textfield for onChange and it needs "event" to get the value of textfield input. 
Even though I use two methods which looks similar in different form or event, can I combine it to one method?

Reply:
refer https://github.com/Cody1009/scoville_todo_app/pull/2/files#r189569005